tests/system/test_cluster/test_agent_groups/test_group_sync_status.py

```diff
@@ -0,0 +1,185 @@
+from system import clean_cluster_logs
+
+pytestmark = [pytest.mark.cluster, pytest.mark.enrollment_cluster_env]
+
+testinfra_hosts = ['wazuh-master', 'wazuh-worker1','wazuh-worker2']
+groups = ['group_master', 'group_worker1', 'group_worker2']
+agents = ['wazuh-agent1', 'wazuh-agent2']
+workers = ['wazuh-worker1', 'wazuh-worker2']
+groups_created = []
+first_time_check = "synced"
+second_time_check = "synced"
+network = {}
+client_keys = {}
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                              'provisioning', 'enrollment_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+test_cases_yaml = read_yaml(os.path.join(local_path, 'data/test_group_sync_cases.yml'))
+wdb_query = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'script/wdb-query.py')
+
+@pytest.fixture
+def delete_logs():
+    clean_cluster_logs(testinfra_hosts, host_manager)
+
+def delete_all_groups():
+    global groups_created
+    groups_created = host_manager.run_command(testinfra_hosts[0], f'{WAZUH_PATH}/bin/agent_groups')
+    for group in groups:
+        if group in groups_created:
+            response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='DELETE',
+                                                      endpoint=f'/groups?groups_list={group}')
+            assert response['status'] == 200, f"Failed to delete {group} group: {response}"
+
+@pytest.fixture
+def group_creation():
+    delete_all_groups()
+    for group in groups:
+        response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='POST',
+                                                      endpoint='/groups', request_body={'group_id': group})
+        assert response['status'] == 200, f"Failed to create {group} group: {response}"
+    
+@pytest.fixture
+def agent_group_assignation():
+    agent_ids = host_manager.run_command(testinfra_hosts[0], f'cut -c 1-3 {WAZUH_PATH}/etc/client.keys').split()
+    for group in groups:
+        for agent_id in agent_ids:
+            response = host_manager.make_api_call(host=testinfra_hosts[0], token=host_manager.get_api_token(testinfra_hosts[0]), method='PUT',
+                                                      endpoint=f'/agents/{agent_id}/group/{group}')
+            assert response['status'] == 200, f"Failed to assign agent {agent_id} in {group} group: {response}"
+    
+@pytest.fixture
+def delete_group_folder(test_case):
+    groups_created = host_manager.run_command(testinfra_hosts[0], f'{WAZUH_PATH}/bin/agent_groups')
+    if test_case['test_case']['group_deleted'] in groups_created:
+        host_manager.run_command(test_case['test_case']['host'], f"rm -r {WAZUH_PATH}/etc/shared/{test_case['test_case']['group_deleted']} -f")
```

Comment:
Check if we can use `delete_agent_groups()`

Reply:
Checked and corrected in https://github.com/wazuh/wazuh-qa/pull/4143